api/include/opentelemetry/logs/logger.h

```diff
@@ -238,18 +239,238 @@ class Logger
    *  span<pair<string_view, AttributeValue>> -> attributes(return type of MakeAttributes)
    */
   template <class... ArgumentType>
-  void Fatal(ArgumentType &&... args) noexcept
+  void Fatal(ArgumentType &&...args) noexcept
   {
     static_assert(
         !detail::LogRecordHasType<Severity, typename std::decay<ArgumentType>::type...>::value,
         "Severity is already set.");
     this->EmitLogRecord(Severity::kFatal, std::forward<ArgumentType>(args)...);
   }
 
+  //
+  // OpenTelemetry C++ user-facing Logs API
+  //
+
+  inline bool Enabled(Severity severity, const EventId &event_id) const noexcept
+  {
+    OPENTELEMETRY_LIKELY_IF(Enabled(severity) == false)
+    {
+      return false;
+    }
+    return EnabledImplementation(severity, event_id);
+  }
+
+  inline bool Enabled(Severity severity, int64_t event_id) const noexcept
+  {
+    OPENTELEMETRY_LIKELY_IF(Enabled(severity) == false)
+    {
+      return false;
+    }
+    return EnabledImplementation(severity, event_id);
+  }
+
+  inline bool Enabled(Severity severity) const noexcept
+  {
+    return static_cast<uint8_t>(severity) >= OPENTELEMETRY_ATOMIC_READ_8(&minimum_severity_);
+  }
+
+  /**
+   * Log an event
+   *
+   * @severity severity of the log
+   * @event_id event identifier of the log
+   * @format an utf-8 string following https://messagetemplates.org/
+   * @attributes key value pairs of the log
+   */
+  virtual void Log(Severity severity,
```

Comment:
Do we need these overloaded functions? We can set the data in the type trait, so we don't have to write a lot of overloaded functions for all parameter variation combinations.

Reply:
@owent, as we are defining user facing API, I'd prefer make the API signature explicit. Type traits could save us all the variant of overloads, but still not all parameter variation combinations makes sense, and we need add more code to the type traits for such validation. Compare with this, probably overload is more straightforward, and doesn't incur runtime overhead (overload resolution happens at compile time). Please let me know what you think about this.

Reply:
Explicit arguments with defined order makes sense for user facing API to me if we have limited number of arguments.